lib/project_types/extension/forms/questions/ask_registration.rb

```diff
@@ -34,8 +34,12 @@ def choose_interactively(apps_and_registrations)
         end
 
         def load_registrations(type)
-          ctx.puts(@ctx.message("connect.loading_extensions"))
-          registrations = Tasks::GetExtensions.call(context: ctx, type: type)
+          registrations = []
+          loading_extensions = @ctx.message("connect.loading_extensions")
+
+          CLI::UI::Spinner.spin(loading_extensions) do |_spinner|
+            registrations = Tasks::GetExtensions.call(context: ctx, type: type)
+          end
```

Comment:
That would be a nice API, but we, unfortunately, [can't](https://github.com/Shopify/cli-ui/blob/9a47e420447bf17006f66f84a073d0c92ca26752/lib/cli/ui/spinner.rb#L67) because spin doesn't return the result of the block.